### PR TITLE
Additional changes related to PR #179 (agent display name).

### PR DESCRIPTION
**Pull request by mfoltzgoogle**
This is a followup to https://github.com/webscreens/openscreenprotocol/pull/179.

It makes the following changes:
- Adds definitions for advertising agent, listening agent, display name, verified display name.
- Clarifies that truncated display names must never be shown to users.
- Clarifies that prefix check is required before the display name is verified.
- Replaced usages of "friendly name" with "display name."  (I could go through and autolink these usages if folks like, but maybe adding autolinks across the spec is better done in a separate PR.)
- Makes the usage of mDNS/DNS-SD a "must." This seems non-controversial to me, as it's the only discovery mechanism at present, but happy to do this in a separate PR if folks want to discuss.

***
<a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/pull/180.html" title="Last updated on Aug 13, 2019, 5:53 PM UTC (3cea78b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/180/b72ed9f...3cea78b.html" title="Last updated on Aug 13, 2019, 5:53 PM UTC (3cea78b)">Diff</a>

### CODE REVIEW

**Comment by mfoltzgoogle** [comment]
PTAL @chrisn @pthatcherg

**Comment by mfoltzgoogle** [comment]
I updated the PR to make name display recommendation a "should" per the F2F minutes.

File: index.bs [hunk 1]
```diff
@@ -280,25 +280,43 @@ Non-Functional Requirements {#requirements-non-functional}
...
-truncated display name.
+An <dfn noexport>advertising agent</dfn> is one that responds to mDNS queries
+for `_openscreen._udp.local`.  Such an agent should have a <dfn noexport>display
+name</dfn> (a non-empty string) that is a human readable description of the
+presentation display, e.g. "Living Room TV."
+
+A <dfn noexport>listening agent</dfn> is one that sends mDNS queries for
+`_openscreen._udp.local`.  Listening agents may have a display name.
+
+Advertising agents must use an DNS-SD instance name that is a prefix of the
```
[the remark this answers is not in the document]
Comment (mfoltzgoogle) [reply index.bs]: Thanks @chrisn.  Capitalized and linked the DNS-SD terms as you suggested, and fixed the typo.

**Comment by chrisn** [comment]
Thanks, Mark.

**Pull request merged**